Mathlib/Analysis/Meromorphic/TrailingCoefficient.lean

```diff
@@ -196,6 +224,22 @@ lemma MeromorphicAt.meromorphicTrailingCoeffAt_mul {f₁ f₂ : 𝕜 → 𝕜} (
       (meromorphicTrailingCoeffAt f₁ x) * (meromorphicTrailingCoeffAt f₂ x) :=
   meromorphicTrailingCoeffAt_smul hf₁ hf₂
 
+/--
+The trailing coefficient of a product is the product of the trailing coefficients.
+-/
+theorem meromorphicTrailingCoeffAt_prod {ι : Type*} {s : Finset ι} {f : ι → 𝕜 → 𝕜} {x : 𝕜}
+    (h : ∀ σ, MeromorphicAt (f σ) x) :
+    meromorphicTrailingCoeffAt (∏ n ∈ s, f n) x = ∏ n ∈ s, meromorphicTrailingCoeffAt (f n) x := by
+  classical
+  apply Finset.induction
+    (motive := fun b' ↦ (meromorphicTrailingCoeffAt (∏ n ∈ b' , f n) x =
+      ∏ n ∈ b', meromorphicTrailingCoeffAt (f n) x))
+  · simp only [Finset.univ_eq_empty, Finset.prod_empty, forall_const]
+    apply meromorphicTrailingCoeffAt_const
+  · intro σ s₁ hσ hind
+    rw [Finset.prod_insert hσ, Finset.prod_insert hσ, (h σ).meromorphicTrailingCoeffAt_mul
+      (MeromorphicAt.prod h), hind]
```

Comment:
The `induction` tactic is nicer here: it guesses the motive for you, with
```
  classical
  induction s using Finset.induction with
  | empty => 
    simp only [Finset.univ_eq_empty, Finset.prod_empty, forall_const]
    apply meromorphicTrailingCoeffAt_const
  | insert σ s₁ hσ hind =>
    rw [Finset.prod_insert hσ, Finset.prod_insert hσ, (h σ).meromorphicTrailingCoeffAt_mul
      (MeromorphicAt.prod h), hind]
```

Reply:
Done.